src/osd/PGLog.h

```diff
@@ -476,18 +476,14 @@ struct PGLog : DoutPrefixProvider {
     }
 
     void index(pg_log_dup_t& e) {
-      if (PGLOG_INDEXED_DUPS) {
```

Comment:
i don't really think we should remove this check just to silence the warning. 

instead i think it should be
```c++
if (indexed_data & PGLOG_INDEXED_DUPS) {
```

@ivancich what do you think?